ApiDoctor.Validation/Http/HttpParser.cs

```diff
@@ -46,38 +48,46 @@ public HttpRequest ParseHttpRequest(string requestString)
                 throw new ArgumentException("requestString was empty or whitespace only. Not a valid HTTP request.");
             }
 
-            StringReader reader = new StringReader(requestString);
+            var mode = ParserMode.FirstLine;
+            var request = new HttpRequest();
+            var reader = new StringReader(requestString);
             string line;
-            ParserMode mode = ParserMode.FirstLine;
-
-            HttpRequest request = new HttpRequest();
-
-            while( (line = reader.ReadLine()) != null)
+            while ((line = reader.ReadLine()) != null)
             {
+                line = line.Trim();
                 switch (mode)
                 {
                     case ParserMode.FirstLine:
+
                         var components = line.Split(' ');
-                        if (components.Length < 2) 
-                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line.");
-                        if (components.Length > 3)
-                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line (more than the expected 3 components)");
 
+                        if (components.Length < 2)
+                            throw new HttpParserRequestException("RequestString does not contain a proper HTTP request first line.");
                         if (components[0].StartsWith("HTTP/"))
                             throw new HttpParserRequestException("RequestString contains an HTTP response.");
 
-                        request.Method = components[0];
-                        request.Url = components[1];
-
-                        if (components.Length >= 3)
+                        string url;
+                        var httpVersion = components[1].StartsWith("HTTP/") ? components[1] : "HTTP/1.1";
```

Comment:
please add a comment describing what the expected input looks like, what the components represent, etc. it can be hard to visualize if you're not familiar

Reply:
Added.